File: CS/2181_Merge_Nodes_in_Between_Zeros.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution {
    public ListNode MergeNodes(ListNode head) {
        ListNode result = new ListNode(0);
        ListNode temp = result;
        int sum = 0;
        head = head.next;
        while(head!=null){
            if(head.val == 0){
                temp.next = new ListNode(sum);
                temp = temp.next;
                sum = 0;
            }
            else{
                sum += head.val;
            }
            head = head.next;
        }
        return result.next;
    }
}

File: Cpp/2181_Merge_Nodes_in_Between_Zeros.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* mergeNodes(ListNode* head) {
        ListNode* result = new ListNode;
        ListNode* temp = result;
        int sum = 0;
        head = head->next;
        
        while(head != nullptr){
            sum += head->val;
            if(head->val == 0){
                temp->next = new ListNode(sum);
                temp = temp->next;
                sum = 0;
            }
            head = head->next;
        }
        return result->next;
    }
};
